### PR TITLE
[SPARK-49713][PYTHON][CONNECT] Make function `count_min_sketch` accept number arguments

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
1, Make function `count_min_sketch` accept number arguments;
2, Make argument `seed` optional;
3, fix the type hints of `eps/confidence/seed` from `ColumnOrName` to `Column`, because they require a foldable value and actually do not accept column name:
```
In [3]: from pyspark.sql import functions as sf

In [4]: df = spark.range(10000).withColumn("seed", sf.lit(1).cast("int"))

In [5]: df.select(sf.hex(sf.count_min_sketch("id", sf.lit(0.5), sf.lit(0.5), "seed")))
...
AnalysisException: [DATATYPE_MISMATCH.NON_FOLDABLE_INPUT] Cannot resolve "count_min_sketch(id, 0.5, 0.5, seed)" due to data type mismatch: the input `seed` should be a foldable "INT" expression; however, got "seed". SQLSTATE: 42K09;
'Aggregate [unresolvedalias('hex(count_min_sketch(id#1L, 0.5, 0.5, seed#2, 0, 0)))]
+- Project [id#1L, cast(1 as int) AS seed#2]
   +- Range (0, 10000, step=1, splits=Some(12))
...
```

### Why are the changes needed?
1, seed is optional in other similar functions;
2, existing type hint is `ColumnOrName` which is misleading since column name is not actually supported

### Does this PR introduce _any_ user-facing change?
yes, it support number arguments

### How was this patch tested?
updated doctests

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: sql/api/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -389,6 +389,18 @@ object functions {
...
+   * Returns a count-min sketch of a column with the given esp, confidence and seed. The result is
+   * an array of bytes, which can be deserialized to a `CountMinSketch` before usage. Count-min
+   * sketch is a probabilistic data structure used for cardinality estimation using sub-linear
+   * space.
+   *
+   * @group agg_funcs
+   * @since 4.0.0
+   */
+  def count_min_sketch(e: Column, eps: Column, confidence: Column): Column =
+    count_min_sketch(e, eps, confidence, lit(SparkClassUtils.random.nextInt))
```
Comment (HyukjinKwon) [inline sql/api/src/main/scala/org/apache/spark/sql/functions.scala:402]: Would need a connect version as well.

**Review by HyukjinKwon — APPROVED** [review]

File: sql/api/src/main/scala/org/apache/spark/sql/functions.scala (hunk 1 above)
Comment (zhengruifeng) [reply sql/api/src/main/scala/org/apache/spark/sql/functions.scala:402]: Good catch! let me add one for Scala client.

Comment (zhengruifeng) [reply sql/api/src/main/scala/org/apache/spark/sql/functions.scala:402]: @HyukjinKwon it seems we don't have a separate `functions.scala` for Scala client

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -6031,26 +6031,67 @@ def count_min_sketch(
...
+            `eps` now accepts float value.
+
+    confidence : :class:`~pyspark.sql.Column` or float
         confidence, must be positive and less than 1.0
-    seed : :class:`~pyspark.sql.Column` or str
+
+        .. versionchanged:: 4.0.0
+            `confidence` now accepts float value.
+
+    seed : :class:`~pyspark.sql.Column` or int
```
Comment (xinrong-meng) [inline python/pyspark/sql/functions/builtin.py:6046]: nit: `, optional` in the end?

**Review by xinrong-meng — APPROVED** [review]

File: python/pyspark/sql/functions/builtin.py (hunk 2 above)
Comment (zhengruifeng) [reply python/pyspark/sql/functions/builtin.py:6046]: nice catch!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by zhengruifeng** [comment]
thanks @dongjoon-hyun @xinrong-meng and @HyukjinKwon 

merged to master

**Pull request closed without merging**